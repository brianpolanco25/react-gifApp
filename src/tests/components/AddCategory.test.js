import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { AddCategory } from '../../components/AddCategory';



describe('Pruebas en el componente <AddCategory />', () => {

    const setCategory = jest.fn(); 
    let wrapper = shallow(<AddCategory setCategories={ setCategory } />);
    

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategory } />);
    })

    test('debe cargar correctamente el componente', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('debe de cambiar la caja de texto', () => {

        const input = wrapper.find('input');
        const inputValue = 'hola mundo';

        input.simulate('change', { target: { inputValue } });
        
    });

    test('NO debe de postear la informacion con submit', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategory).not.toHaveBeenCalled();

    })

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {

        const value = 'hola mundo';

        wrapper.find('input').simulate('change', { target: { value } });

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategory).toHaveBeenCalledTimes(1);

        expect(setCategory).toHaveBeenCalledWith(expect.any(Function));

        expect( wrapper.find('input').prop('value') ).toBe('');

    })
    
    
    
    
    
})
