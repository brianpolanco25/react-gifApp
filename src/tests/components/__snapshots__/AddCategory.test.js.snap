// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory /> debe cargar correctamente el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
